shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-rewrite/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/ProjectionsToken.java

```diff
@@ -18,26 +18,29 @@
 package org.apache.shardingsphere.sharding.rewrite.token.pojo;
 
 import org.apache.shardingsphere.infra.rewrite.sql.token.pojo.Attachable;
+import org.apache.shardingsphere.infra.rewrite.sql.token.pojo.RouteUnitAware;
 import org.apache.shardingsphere.infra.rewrite.sql.token.pojo.SQLToken;
+import org.apache.shardingsphere.infra.route.context.RouteUnit;
 
 import java.util.Collection;
+import java.util.Map;
 
 /**
  * Projections token.
  */
-public final class ProjectionsToken extends SQLToken implements Attachable {
+public final class ProjectionsToken extends SQLToken implements Attachable, RouteUnitAware {
     
-    private final Collection<String> projections;
+    private final Map<RouteUnit, Collection<String>> projections;
     
-    public ProjectionsToken(final int startIndex, final Collection<String> projections) {
+    public ProjectionsToken(final int startIndex, final Map<RouteUnit, Collection<String>> projections) {
```

Comment:
The object Map<RouteUnit, Collection<String>> it too hard to understand, can you consider about to create a new object to describe it?

Reply:
The struct just a result for all RouteUnit, different routeUnit correspond to different rewrited projection, if create a new object,it maybe a little heavy.

Reply:
t too hard to understand, what is the Collection<String>?
It is better to make the code for read easier.

Reply:
The original struct is `Collection<String> projections`,it's a list of rewrited derived projection string,such as `order by`, how about add some comments?
![image](https://user-images.githubusercontent.com/16559542/93414047-1dcd4080-f8d3-11ea-97e9-3e916c67b381.png)